daemon/api.go

```diff
@@ -893,6 +910,9 @@ func postSnap(c *Command, r *http.Request, user *auth.UserState) Response {
 
 	msg, tsets, err := impl(&inst, state)
 	if err != nil {
+		if _, ok := err.(reqerr); ok {
```

Comment:
I suggest dropping these reqerr uses and changing the InternalError below to BadRequest instead. I think we're incorrectly assuming that anything below this depth is an internal error by default, but looking through this API it's clear that the vast majority of errors we see are actually constraints imposed on the request rather than internal problems (pretty much everything on this function is a BadRequest.. why would things inside impl not be?).